crates/database/migration/src/m20250408_150338_load_header_metadata.rs

```
@@ -270,23 +277,46 @@ impl MetadataDecoder {
         let flag = buf[0];
         let vanity_index = buf[1];
 
+        let has_coinbase = (flag & 0b00010000) != 0;
+        let has_nonce = (flag & 0b00100000) != 0;
         let difficulty = if flag & 0b01000000 == 0 { 2 } else { 1 };
         let seal_length = if flag & 0b10000000 == 0 { 65 } else { 85 };
         let vanity = self.vanity.get(&vanity_index)?;
 
-        if buf.len() < B256::len_bytes() + seal_length + 2 {
+        // flag + vanity index + state root + coinbase + nonce + seal
+        let total_expected_size = 2 * size_of::<u8>() +
+            B256::len_bytes() +
+            Address::len_bytes() * has_coinbase as usize +
+            size_of::<u64>() * has_nonce as usize +
+            seal_length;
+
+        if buf.len() < total_expected_size {
```

Should we raise an error in this case?